web/pdf_find_controller.js

```diff
@@ -120,16 +164,41 @@ var PDFFindController = (function PDFFindControllerClosure() {
         query = query.toLowerCase();
       }
 
-      var matches = [];
-      var matchIdx = -queryLen;
-      while (true) {
-        matchIdx = pageContent.indexOf(query, matchIdx + queryLen);
-        if (matchIdx === -1) {
-          break;
+      if (phraseSearch) {
+        subqueryArray = [query];
+      } else {
+        subqueryArray = query.split(DIVIDER_TERMS);
+      }
+      subqueryArray.forEach(function(subquery){
+        if (subquery === '') {
+          return;
+        }
+        var query = subquery;
+        var queryLen = query.length;
+        var matchIdx = -queryLen;
+        while (true) {
+          matchIdx = pageContent.indexOf(query, matchIdx + queryLen);
+          if (matchIdx === -1) {
+            return;
+          }
+          if (!phraseSearch) {
+            matchesWithLength.push({ match: matchIdx, matchLength: queryLen });
```

Comment:
Since this array is not needed for "normal" searching, you could initialize it to `null` above. Then simply add the following code before the current line:
```javascript
if (!matchesWidthLength) {
  matchesWidthLength = [];
}
```
This way it's lazily initialized instead.